src/librbd/io/QosImageDispatch.cc

```diff
@@ -279,6 +284,7 @@ bool QosImageDispatch<I>::needs_throttle(
   auto cct = m_image_ctx->cct;
   auto extent_length = get_extent_length(image_extents);
   bool all_qos_flags_set = false;
+  auto flags_to_set = 0;
```

Comment:
Nit: It's better to avoid auto with integer types, particularly on initialization as the variable ends up as `int` and then sign mismatch warnings can ensue:
```suggestion
  uint32_t flags_to_set = 0;
```